SpotGamma.cpp

```diff
@@ -1,249 +1,284 @@
-// The stdlib includes are included for us in the SierraChart "remote build"
-// #include <iostream>
-// #include <string>
 #include "sierrachart.h"
+#include <string>
+#include <sstream>
+#include <vector>
+SCDLLName("SpotGamma Levels Importer")
+
+/*
+
+    The format of the CSV is as follows:
+        (float)  Price
+        (string) Note
+        (string) Color
+        
+*/
+
+// Struct we'll use to reference our vector
+struct PriceLabel {
+    float Price;
+    SCString Label;
+    COLORREF LabelColor;
+};
+
+SCSFExport scsf_SpotGammaLevels(SCStudyInterfaceRef sc)
+{
+    const SCString FAKE_URL = "https://api.spotgamma.com/integration/FAKE_URL/SPX_sierra.csv";
+    
+    // logging object
+    SCString msg;
+
+    // General Input Settings
+    SCInputRef UrlPath = sc.Input[0];
+    SCInputRef DrawLinesOnChart = sc.Input[2];
+    SCInputRef ShowPriceOnChart = sc.Input[3];
+    // Update Interval Input (in minutes)
+    SCInputRef UpdateInterval = sc.Input[4];
+    
+    // Set configuration variables
+    if (sc.SetDefaults)
+    {
+        sc.GraphName = "SpotGamma Levels";
+        sc.GraphRegion = 0;
+
+        UrlPath.Name = "CSV URL";
+        // example template:
+        // https://docs.google.com/spreadsheets/d/1fx_Ym6QodEJIqQO90HR1dWYFw5RoK2w5ywtHeMlMeZY/gviz/tq?tqx=out:csv
+        // Google Docs Spreadsheet
+        // REQUIRED:
+        //    - set SHARING to "Anyone with link"
+        //    - strip off anything after the unique key, (eg "/edit#usp=sharing")
+        UrlPath.SetString(FAKE_URL);
+
+        // Chart specific settings
+        DrawLinesOnChart.Name = "Draw Lines?";
+        DrawLinesOnChart.SetYesNo(1);
+
+        ShowPriceOnChart.Name = "Show Price Label?";
+        ShowPriceOnChart.SetYesNo(0);
+
+        UpdateInterval.Name = "Update Interval (in Min, 0 = manual update with INS key)";
+        UpdateInterval.SetInt(0);
+
+        return;
+    }
 
-SCDLLName("SpotGamma")
-
-const SCString FAKE_URL = "https://spotgamma-system-files.s3.amazonaws.com/ENTER_VALID_URL.csv";
-
-/**************
- * Constants  *
- **************/
+    // Process to initialize variables and pointers along with StateMessage for debug
+    int UpdateIntervalInMin = UpdateInterval.GetInt();
+    int UpdateIntervalInSec = UpdateIntervalInMin * 60;
+    int &LastUpdated = sc.GetPersistentInt(9);
+    int& RequestState = sc.GetPersistentInt(1);
```

Comment:
nit: let's get these refs consistent:
```suggestion
    int &RequestState = sc.GetPersistentInt(1);
```